products.js

```diff
@@ -197,44 +197,242 @@ function tunedTitleFnodes(nodeToCssMap) {
     return tuningRoutine;
 }
 
-function tunedPriceFnodes(nodeToCssMap, coeffDollarSign = 1, coeffCSS = 1, coeffSavings = 0.5) {
+
+function tunedPriceFnodes(nodeToCssMap, coeffDollarSign = 1.6, coeffNearDollarSign = 1, coeffHasNumbers = 0.7,
+  coeffSpanBonus = 0.7, coeffSemanticTags = 1.9000000000000001, coeffCurrentPrice = 0.39999999999999997, coeffItemprop = 1, coeffKeywords = 1.3,
+  coeffStrike = 0.7, coeffNotSavings = 1.3, coeffAboveFold = 1.3, coeffCenterRight = 1.3, coeffMiddleHeight = 0.09999999999999998,
+  coeffBolded = 1, coeffNumNumbers = 0.39999999999999997, coeffNumDollarSigns = 0.39999999999999997, coeffPriceFormat = 0.7, coeffNumDots = 1.9000000000000001, coeffMetaTag = 7.3) {
 
     function hasDollarSign(fnode){
-      if(fnode.element.childNodes[0] && fnode.element.childNodes[0].nodeValue && fnode.element.childNodes[0].nodeValue.includes('$')){
+      if(fnode.element.textContent.includes('$')){
```

Comment:
You know, you wrote this ruleset in a way that I never would have dared. And it totally worked! You used expensive constructs like .textContent (which has to iteratate over a whole subtree and collect the text) with abandon, and you got a very high score. My congratulations!

But that does bring to mind another thing you can do (since you were looking): bench your rulesets and see if they take more than a couple milliseconds. If they do and there's an evident way to speed them up, do so. That's another point of reluctance I've heard in some quarters about smarter browser behavior.